cmd/oras/internal/option/remote.go

```diff
@@ -243,6 +251,21 @@ func (opts *Remote) authClient(registry string, debug bool) (client *auth.Client
 	return
 }
 
+func (opts *Remote) parseCustomHeaders() error {
+	if len(opts.headerFlags) != 0 {
+		headers := map[string][]string{}
+		for _, h := range opts.headerFlags {
+			name, csv, found := strings.Cut(h, ":")
+			if !found || strings.TrimSpace(csv) == "" {
+				return fmt.Errorf("cannot parse headers: %q", h)
+			}
+			headers[name] = append(headers[name], strings.Split(csv, ",")...)
```

Comment:
No need to unbox the value, just add it to the http headers.
```suggestion
			name, value, found := strings.Cut(h, ":")
			if !found || name == "" || strings.TrimSpace(value) == "" {
				return fmt.Errorf("invalid header: %q", h)
			}
			headers[name] = append(headers[name], value)
```

Reply:
key can be empty string in curl.

Reply:
removed the unboxing.